packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -45,12 +40,91 @@ export const useSwitch = (props: SwitchProps): SwitchInfo => {
   const onChangeWithAnimation = React.useCallback(
     (e: InteractionEvent, checked?: boolean) => {
       onChange && onChange(e, checked);
-      animateSwitchKnob();
+      if (isMobile) {
```

Comment:
With line 118 this seems like it would run more often than it should, we should probably remove the duplicate. I think running the animation in a useEffect hook would be best since that would cover controlled switches, too.

Reply:
Moved into useEffect() , animation runs good.